util/tls/tls_socket.cc

```diff
@@ -326,10 +326,23 @@ SSL* TlsSocket::ssl_handle() {
 }
 
 auto TlsSocket::MaybeSendOutput() -> error_code {
-  // this function is present in both read and write paths.
+  // This function is present in both read and write paths.
   // meaning that both of them can be called concurrently from differrent fibers and then
   // race over flushing the output buffer. We use state_ to prevent that.
   if (state_ & WRITE_IN_PROGRESS) {
+    // Do not remove the Yield from here because it can enter an infinite loop
+    // which can be described by the following chore:
+    // Fiber1 (Connection fiber):
+    // Reads SSL renegotiation request
+    // Sets WRITE_IN_PROGRESS in state_
+    // Tries to write renegotiation response to socket
+    // BIO (SSL buffer) gets filled up
+    // Fiber2 (runs command):
+    // Tries to write response to socket
+    // Can't write to socket because WRITE_IN_PROGRESS is set
+    // Live blocks indefinitely
```

Comment:
Most probably it fixes the issue but we do not have much visibility to these spinning behaviors inside `TlsSocket::RecvMsg` and `TlsSocket::SendBuffer`.
it seems we can identify them. For example, count number of consecutive loops that `op_val == 0` inside  SendBuffer and if it crosses some hardcoded limit, issue an warning.  maybe you can think of something more elegant. in any case I want to see more logs around these loops